### PR TITLE
#4587: Reversed isAssignableFrom arguments

### PR DESCRIPTION
**Pull request by misl**
QuarkusMediatorConfigurationUtil.JandexGenericTypeAssignable.check() results in NotAssignable where Assignable is expected. Especially the following part: argumentClass.isAssignableFrom(target). This basically results in AmqpMessage.isAssignableFrom(Message) or MqttMessage.isAssignableFrom(Message). This should be the other way around Message.isAssignableFrom(AmqpMessage).

### CODE REVIEW

**Comment by geoand** [comment]
@cescoffier How do you want to deal with the testing part of this PR? Should we add a full blown integration test in the `integration-tests` directory?

**Comment by geoand** [comment]
Can you please rebase your PR onto the latest master in order to pick up a fix for CI?

Thanks

**Comment by cescoffier** [comment]
We would need another PR with tests. Test would cover this but also other usages.

**Comment by cescoffier** [comment]
So once rebased, and if CI is green, let's merge that and open another PR with tests.

**Review by geoand — APPROVED** [review]
Thanks for this! Let's wait for CI and merge if all is green

**Comment by geoand** [comment]
@misl are you sure you rebased to the latest master? The native security tests are still failing and I am not seing that failure in other PRs

**Pull request merged**
